packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -115,14 +152,23 @@ export function CrossmintWalletProvider({
 
     const onAuthRequired = async (
         needsAuth: boolean,
-        sendEmailWithOtp: () => Promise<void>,
+        sendMessageWithOtp: () => Promise<void>,
         verifyOtp: (otp: string) => Promise<void>,
         reject: () => void
     ) => {
-        setEmailSignerDialogOpen(needsAuth);
+        // Check if we're dealing with a phone signer
+        if (createOnLogin?.signer.type === "phone" && createOnLogin.signer.phone) {
+            setPhoneNumber(createOnLogin.signer.phone);
```

Comment:
what if we set the phone via login using setCustomAuth? Shouldnt we use a similar approach to email where instead of setting here, we just fetch it from either customAuth or createOnLogin?

Reply:
indeed, updated